pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCAccessor.java

```diff
@@ -131,15 +139,20 @@ private void buildExpression(SearchArgument.Builder builder, List<Object> filter
                         builder.startNot();
                         break;
                 }
-                buildExpression(builder, ((LogicalFilter) f).getFilterList());
-                builder.end();
+                if (buildExpression(builder, ((LogicalFilter) f).getFilterList()))
+                    builder.end();
+                else
+                    return false;
             } else {
-                buildArgument(builder, f);
+                if (!buildArgument(builder, f))
```

Comment:
We need to return false only if buildArgument(builder, f) call returned false, in case of true we want to proceed.